src/pokemonsuper.java

```diff
@@ -0,0 +1,70 @@
+import java.util.Scanner;
+
+public abstract class pokemonsuper {
+    private String name;
+    private int level;
+    private int hp;
+    private String food;
+
+
+    public pokemonsuper(String name, int level, int hp, String food) {
+        this.name = name;
+        this.level = level;
+        this.hp = hp;
+        this.food = food;
+    }
+
+    public pokemonsuper() {
+    }
+
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public int getLevel() {
+        return level;
+    }
+
+    public void setLevel(int level) {
+        this.level = level;
+    }
+
+    public int getHp() {
+        return hp;
+    }
+
+    public void setHp(int hp) {
+        this.hp = hp;
+    }
+
+    public String getFood() {
+        return food;
+    }
+
+    public void setFood(String food) {
+        this.food = food;
+    }
+
+    public void pokemonLevel() {
+        System.out.println("Your pokemon's level is" + " " + getLevel());
+    }
+
+    public void levelUp() {
+        this.level++;
+        System.out.println("Your" + " " + name + " " + "leveled up! New level is:" + level);
+    }
+
+    public void pokemonName() {
```

Comment:
Deze methode vraagt letterlijk om de naam voor een nieuwe pokemon. Daarom kun je misschien beter garanderen datdeze methode alleen wordt aangeroepen als je een nieuwe pokemon maakt, dus in de constructor. Je kunt het dan private maken en expliciet aanroepen in de constructor. In de lege constructor kun je deze methode gewoon aanroepen. In de andere constructor kun je deze methode aanroepen in plaats van `this.name=name` en kun je de `name` parameter laten vervallen.